docs: Add Y-Sweet

jakelazaroff opened the pull request:
This PR adds [Y-Sweet](https://jamsocket.com/y-sweet) to…

- …the [Real-time Collaborative rich text editor](https://www.blocknotejs.org/docs/advanced/real-time-collaboration) page in the docs.
- …the Collaboration section of the examples. Note that because Y-Sweet requires a server-side endpoint, we omitted the example embedded within the docs and linked to our example hosted on [demos.y-sweet.dev](https://demos.y-sweet.dev/blocknote?doc=ihvp6vi0u0j).

Also fixes a typo on the Liveblocks and PartyKit examples 🙂 

Also also heads up that I'm getting a bunch of `docs: Module not found: Can't resolve '@blocknote/core'` errors whenever I try to run `npm run build:site`!

YousefED reviewed (commented):
Thanks! I removed the stuff in caps, a bit too much (for now).

Two questions:
- While you link to a live demo, I'm missing example code or blocknote-specific documentation on how to set it up (even on the links you added). Are these WIP?
- I understand your idea now is that you didn't add a live demo as it requires a collab server. Nevertheless, I think a live demo would be great, also because our examples are all expected to be live examples. Is it possible to (a) set up a dev server for this purpose, or (b) let the user enter a URL of their jamsocket / y-sweet instance? (strong preference for (a)

jakelazaroff commented:
@YousefED We set up a dev server! The example now points to `demos.y-sweet.dev`.

As to your question about the link, I switched the links to more closely mimic the Liveblocks and PartyKit ones (and removed the link to the live demo on our site, since it's embedded in the page now). Right now it points to the Y-Sweet getting started page, but we should have a tutorial and example repo specifically for getting started with BlockNote in the next day or so.

jakelazaroff commented:
Hi @YousefED, this should be fully ready now! The link in the Y-Sweet example now links to this tutorial https://docs.jamsocket.com/y-sweet/tutorials/blocknote
